### ISSUE TITLE
Long inference time while using flag_gems.use_gems()

### ISSUE THREAD

**Issue by lirui635693142** [FlagOpen/FlagGems#354]
### Please describe your question

While testing model accuracy, I found that the inference time of flag_gems.use_gems() is extremely higher than that in torch.no_grad().

**Comment by iclementine** [comment]
The main reason for this is the the Autotuner, which tunes for the best config when it encounters inputs with new key.

We are working on solving this. Possible solutions are
1. Use heuristics instead of Autotuner;
2. use smaller config space;
3. Use the int(log2(size)) instead of size as Autotuner keys;
4. Use an Autotuner with persistent cache.

**Comment by lirui635693142** [comment]
Thxs a lot!

**Comment by BuaaAlban** [comment]
Hi could you give me a example about how to modify it?


```
import torch
import triton
import triton.language as tl
import time
import numpy as np
import sys
import os

# 添加项目根目录到路径
sys.path.append(os.path.dirname(os.path.dirname(os.path.dirname(os.path.abspath(__file__)))))

from src.ops.cumsum import cumsum

# Import flag_gems and set HAS_FLAGGEMM
try:
    import flag_gems
    HAS_FLAGGEMM = True
except ImportError:
    HAS_FLAGGEMM = False

def benchmark_different_sizes():
    """对不同大小的张量进行手动基准测试"""
    sizes = [
        (32, 32),
        (64, 64),
        (128, 128),
        (256, 256),
        (512, 512)
    ]
    results = []
    default_dtype = torch.float32
    dim_to_test = 1
    num_reps = 50

    print(f"\n===== Cumsum: Different Sizes (Manual, dtype={default_dtype}, dim={dim_to_test}) =====")
    header = f"{'Size':<15} {'PyTorch (ms)':<15} {'FlagGEMM (ms)':<18}"
    print(header)
    print("-" * len(header))
    
    for current_size_tuple in sizes:
        x = torch.randn(current_size_tuple, device='cuda', dtype=default_dtype)
        
        # Warmup
        for _ in range(10):
            pytorch_output = torch.cumsum(x, dim=dim_to_test)
            if HAS_FLAGGEMM:
                try:
                    x_fg_warmup = x.clone().to(device=flag_gems.device)
                    with flag_gems.use_gems():
                        torch.cumsum(x_fg_warmup, dim=dim_to_test)
                except Exception: pass
            torch.cuda.synchronize()
        
        # Test PyTorch
        torch.cuda.synchronize()
        start_pytorch = time.time()
        for _ in range(num_reps):
            pytorch_output = torch.cumsum(x, dim=dim_to_test)
            torch.cuda.synchronize()
        pytorch_time_ms = (time.time() - start_pytorch) * 1000 / num_reps
        
        # Test FlagGEMM
        flaggemm_time_ms = float('inf')
        if HAS_FLAGGEMM:
            try:
                x_fg = x.clone().to(device=flag_gems.device)
                torch.cuda.synchronize(device=flag_gems.device if hasattr(flag_gems, 'device') and flag_gems.device is not None else None)
                start_flaggemm = time.time()
                with flag_gems.use_gems():
                    for _ in range(num_reps):
                        flaggemm_output_fg_device = torch.cumsum(x_fg, dim=dim_to_test)
                        torch.cuda.synchronize(device=flag_gems.device if hasattr(flag_gems, 'device') and flag_gems.device is not None else None)
                flaggemm_time_ms = (time.time() - start_flaggemm) * 1000 / num_reps
                with flag_gems.use_gems():
                    flaggemm_output_cpu = torch.cumsum(x_fg, dim=dim_to_test).cpu()
            except Exception:
                flaggemm_time_ms = float('inf')
        
        # Validation
        # assert torch.allclose(triton_output, pytorch_output, atol=1e-5, rtol=1e-3)
        # if flaggemm_output_cpu is not None and flaggemm_time_ms != float('inf'):
        #     assert torch.allclose(pytorch_output.cpu(), flaggemm_output_cpu, atol=1e-5, rtol=1e-3)


        flaggemm_ms_str = f"{flaggemm_time_ms:.3f}" if flaggemm_time_ms != float('inf') else "N/A"

        results.append({
            'size': str(current_size_tuple),
            'pytorch_ms': pytorch_time_ms,
            'flaggemm_ms': flaggemm_time_ms
        })
        print(f"{str(current_size_tuple):<15} {pytorch_time_ms:<15.3f} {flaggemm_ms_str:<18}")
    
    return results

benchmark_different_sizes()

===== Cumsum: Different Sizes (Manual, dtype=torch.float32, dim=1) =====
Size            PyTorch (ms)    FlagGEMM (ms)     
--------------------------------------------------
(32, 32)        0.017           0.106             
(64, 64)        0.039           0.110             
(128, 128)      0.020           0.108             
(256, 256)      0.025           0.106 
```
here is my code 
and the result is alow slow than your perf fig(3x  and my test result os only 0.2 x)
